CMark.hsc

```diff
@@ -106,20 +153,42 @@ nodeToX renderer opts mbWidth node = Unsafe.unsafePerformIO $ do
 
 commonmarkToX :: Renderer
               -> [CMarkOption]
+              -> [CMarkExtension]
               -> Maybe Int
               -> Text
               -> Text
-commonmarkToX renderer opts mbWidth s = Unsafe.unsafePerformIO $
+commonmarkToX renderer opts exts mbWidth s = Unsafe.unsafePerformIO $
   TF.withCStringLen s $ \(ptr, len) -> do
     let opts' = combineOptions opts
-    nptr <- c_cmark_parse_document ptr len opts'
+    exts' <- fmap catMaybes $ mapM resolveExt exts
+    parser <- c_cmark_parser_new opts'
+    mapM_ (c_cmark_parser_attach_syntax_extension parser) exts'
+    c_cmark_parser_feed parser ptr len
+    nptr <- c_cmark_parser_finish parser
+    c_cmark_parser_free parser
     fptr <- newForeignPtr c_cmark_node_free nptr
     withForeignPtr fptr $ \p -> do
       str <- renderer p opts' (fromMaybe 0 mbWidth)
       t <- TF.peekCStringLen $! (str, c_strlen str)
       return t
 
-type NodePtr = Ptr ()
+data ParserPhantom
+type ParserPtr = Ptr ParserPhantom
+
+data NodePhantom
+type NodePtr = Ptr NodePhantom
+
+data LlistPhantom a
+type LlistPtr a = Ptr (LlistPhantom a)
+
+data MemPhantom
+type MemPtr = Ptr MemPhantom
+
+data PluginPhantom
+type PluginPtr = Ptr PluginPhantom
+
+data ExtensionPhantom
+type ExtensionPtr = Ptr ExtensionPhantom
```

Comment:
Added all these phantom `data`s to parameterise our `Ptr` types with because I was mixing up pointer types somewhere and I wasn't being warned since they were all `Ptr ()`.